src/net.cpp

```diff
@@ -2034,6 +2037,11 @@ void CConnman::ThreadOpenConnections(const std::vector<std::string> connect)
                 addr = addrman.Select();
             }
 
+            // The following checks are never applied to MANUAL and ADDR_FETCH,
+            // because those connections are made elsewhere.
+            assert(conn_type != ConnectionType::MANUAL);
+            assert(conn_type != ConnectionType::ADDR_FETCH);
+
```

Comment:
I'm not sure how this is an improvement. `conn_type` is a local variable that is set in lines 1932-1988 above, and can never be set to `ConnectionType::MANUAL` or `ConnectionType::ADDR_FETCH`. What are these assertions preventing or protecting against?